From 1f00f87ce7571ca9e85296fce5144f510cd7722b Mon Sep 17 00:00:00 2001
From: Ben Hutchings <ben@decadent.org.uk>
Date: Fri, 25 Mar 2011 02:57:02 +0000
Subject: [PATCH 1/2] btrfs: Disable FS_IOC_FIEMAP ioctl

btrfs does not implement the fiemap operation correctly for extents
that are subject to delayed allocation.  This has been fixed upstream,
but would be tough to backport.
---
 fs/btrfs/inode.c |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/fs/btrfs/inode.c b/fs/btrfs/inode.c
index e03a836..bd10c29 100644
--- a/fs/btrfs/inode.c
+++ b/fs/btrfs/inode.c
@@ -4908,7 +4908,11 @@ static ssize_t btrfs_direct_IO(int rw, struct kiocb *iocb,
 static int btrfs_fiemap(struct inode *inode, struct fiemap_extent_info *fieinfo,
 		__u64 start, __u64 len)
 {
+#if 0
 	return extent_fiemap(inode, fieinfo, start, len, btrfs_get_extent);
+#else
+	return -EOPNOTSUPP;
+#endif
 }
 
 int btrfs_readpage(struct file *file, struct page *page)
-- 
1.7.4.1

